cunumeric/bits.py

```diff
@@ -0,0 +1,181 @@
+        `packed` has the same number of dimensions as the input (unless `axis`
+        is None, in which case the output is 1-D).
+
+    See Also
+    --------
+    numpy.packbits
+
+    Availability
+    --------
+    Multiple GPUs, Multiple CPUs
+    """
+
+    a, sanitized_axis = _sanitize_arguments(a, axis, bitorder)
+
+    if a.dtype.kind not in ("u", "i", "b"):
+        raise TypeError(
+            "Expected an input array of integer or boolean data type"
+        )
+
+    out_shape = tuple(
+        (extent + 7) // 8 if dim == sanitized_axis else extent
+        for dim, extent in enumerate(a.shape)
+    )
+    out = empty(out_shape, dtype="B")
+    out._thunk.packbits(a._thunk, sanitized_axis, bitorder)
+
+    return out
+
+
+@add_boilerplate("a")
+def unpackbits(
+    a: ndarray,
+    axis: Optional[int] = None,
+    count: Optional[int] = None,
+    bitorder: str = "big",
+) -> ndarray:
+    """
+    Unpacks elements of a uint8 array into a binary-valued output array.
+
+    Each element of `a` represents a bit-field that should be unpacked
+    into a binary-valued output array. The shape of the output array is
+    either 1-D (if `axis` is ``None``) or the same shape as the input
+    array with unpacking done along the axis specified.
+
+    Parameters
+    ----------
+    a : ndarray[uint8]
+       Input array.
+    axis : int, optional
+        The dimension over which bit-unpacking is done.
+        ``None`` implies unpacking the flattened array.
+    count : int or None, optional
+        The number of elements to unpack along `axis`, provided as a way
+        of undoing the effect of packing a size that is not a multiple
+        of eight. A non-negative number means to only unpack `count`
+        bits. A negative number means to trim off that many bits from
+        the end. ``None`` means to unpack the entire array (the
+        default). Counts larger than the available number of bits will
+        add zero padding to the output. Negative counts must not
+        exceed the available number of bits.
+
+    bitorder : ``{"big", "little"}``, optional
+        The order of the returned bits. 'big' will mimic bin(val),
+        ``3 = 0b00000011 => [0, 0, 0, 0, 0, 0, 1, 1]``, 'little' will reverse
+        the order to ``[1, 1, 0, 0, 0, 0, 0, 0]``.
+        Defaults to 'big'.
+
+    Returns
+    -------
+    unpacked : ndarray[uint8]
+       The elements are binary-valued (0 or 1).
+
+    See Also
+    --------
+    numpy.unpackbits
+
+    Availability
+    --------
+    Multiple GPUs, Multiple CPUs
+    """
+
+    a, sanitized_axis = _sanitize_arguments(a, axis, bitorder)
+
+    if a.dtype != "B":
+        raise TypeError("Expected an input array of unsigned byte data type")
+
+    if count is None:
+        sanitized_count = a.shape[sanitized_axis] * 8
+    else:
+        if not isinstance(count, int):
+            raise TypeError("count must be an integer")
+        sanitized_count = count
```

Comment:
these else logic can be replaced with the line 172 since count is not supported in this implementation

Reply:
`count` is now handled correctly (4cd1a68304b165619d211a3389b04ea197c270e7)